app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -223,30 +223,38 @@ function ($attribute, $value, $fail) {
             return response()->json(['error' => 'Failed to convert KML to GeoJSON', 'message' => $process->getErrorOutput()], 500);
         }
         $geojsonContent = file_get_contents($geojsonPath);
-        $redis_key = 'kml_file_' . uniqid();
-        Redis::set($redis_key, $geojsonContent, 'EX', 7200);
+        $polygonLoadedList = isset($body['polygon_loaded']) && 
+            filter_var($body['polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
+        $submitPolygonsLoaded = isset($body['submit_polygon_loaded']) && 
+            filter_var($body['submit_polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
     
-        $polygonLoadedList = isset($body['polygon_loaded']) && filter_var($body['polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
-        $submitPolygonsLoaded = isset($body['submit_polygon_loaded']) && filter_var($body['submit_polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
-
-        $delayedJob = DelayedJob::create(['status' => InsertGeojsonToDBJob::STATUS_PENDING]);
-
-        if (! $polygonLoadedList && !$submitPolygonsLoaded) {
-          $job = new InsertGeojsonToDBJob($redis_key, $delayedJob->id, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
-        }
-
         if ($polygonLoadedList) {
-          $polygonLoaded = $this->GetAllPolygonsLoaded($geojsonContent, $site_id);
-          return response()->json($polygonLoaded->original, 200);
+            $polygonLoaded = $this->GetAllPolygonsLoaded($geojsonContent, $site_id);
+            return response()->json($polygonLoaded->original, 200);
         }
 
-        if ($submitPolygonsLoaded) {
-          $job = new InsertGeojsonToDBJob($redis_key, $delayedJob->id, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
+        $redis_key = 'kml_file_' . uniqid();
+        Redis::set($redis_key, $geojsonContent, 'EX', 7200);
+        $delayedJob = DelayedJob::create(['status' => DelayedJob::STATUS_PENDING]);
+    
+        if($delayedJob) {
```

Comment:
I think this check is superfluous. There isn't a reasonable expectation that creating a row in this DB table could fail (and please fix in the lower instances of this pattern as well)

Reply:
removed